[XSkull] Support v1.21.1b78+ ResolvableProfile

rexlManu opened the pull request:
Already sometime the ResolvableProfile exists (I think since 1.20.5) but now the GameProfile was replaced with the ResolvableProfile both in the CraftSkull & CraftMetaSkull classes. ResolvableProfile has a constructor accepts only a GameProfile to create a profile from a GameProfile and it also saves a gameProfile that is also created when it's not created with a game profile.

This pr adds two new method handles. One for creating a ResolvableProfile from a GameProfile and one to get the GameProfile back from a ResolvableProfile. I tested myself the changes for my plugins and with my changes it, Skulls (both Items & Blocks) working again on paper build 83.

This would also fix #298

_If I'm honest, I have zero clue how your abstraction over reflection even works and it looks really "magically" and I'm sure I didn't properly followed the current code style, so if you want to implement yourself, just use this pr to see how it works._

CryptoMorin commented:
Thank you so much. The way you used the reflection is actually correct. There are just small semantic issues, but that's fine, I can change those later.

Are you able to test this in v1.21.1 build 50, v1.21.0, and v1.20.6 just to be sure they work correctly?

rexlManu commented:
Alright I tested it with paper build 50, 1.21 (doesn't exists anymore since it was replaced with 1.21.1 but still tested it tho), 1.20.6 and 1.20.4. I found a issue on 1.20.4 and older and changed how I check the type. I assumed ReflectiveHandle#exists, covers the exception but the exception is still thrown. I replaced it with a inner try catch. I don't like that approach but I'm just to unfamiliar with the reflection abstraction, so that needs abit of touch.

rexlManu commented:
What I should mention is that I only tested it with paper. The only different for spigot is that the gameProfile() method is [called f() for now (1.20.6-1.21.1)](https://mappings.dev/1.21.1/net/minecraft/world/item/component/ResolvableProfile.html)
